### PR TITLE
cleanup(pubsub): move client samples to generated admin api

### PR DESCRIPTION
**Pull request by alevenberg**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13534)

### CODE REVIEW

File: google/cloud/pubsub/samples/client_samples.cc [hunk 1]
```diff
@@ -161,12 +161,13 @@ void TopicAdminClientSetEndpoint(std::vector<std::string> const& argv) {
     throw examples::Usage{"topic-admin-client-set-endpoint"};
   }
   //! [topic-admin-client-set-endpoint]
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/client_samples.cc:163]: We should delete these handwritten samples and point https://github.com/googleapis/google-cloud-cpp/blob/115a348560952928e13ed3ad2f28488a3d827ba4/google/cloud/pubsub/doc/pubsub-main.dox#L123 to https://github.com/googleapis/google-cloud-cpp/blob/115a348560952928e13ed3ad2f28488a3d827ba4/google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc#L40

for all of the generated admin snippets.

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: google/cloud/pubsub/samples/client_samples.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/client_samples.cc:163]: Does deleting the handwritten ones need to be done separately?

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 2]
```diff
@@ -102,31 +102,31 @@ Follow these links to find examples for other `*Client` classes:
 
 /*! @page Publisher-endpoint-snippet Override @c Publisher Default Endpoint
 
-@snippet google/cloud/pubsub/samples/client_samples.cc publisher-set-endpoint
+@snippetgoogle/cloud/pubsub/admin/samples/topic_admin_client_samples.cc publisher-set-endpoint
```
Comment (dbolduc) [inline google/cloud/pubsub/doc/pubsub-main.dox:105]: I imagine the `publish-docs` build will fail. We can only move the generate admin samples. (The ones with `TopicAdminClient`, `SubscriptionAdminClient`)

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 3]
```diff
@@ -102,31 +102,31 @@ Follow these links to find examples for other `*Client` classes:
...
 
-@snippet google/cloud/pubsub/samples/client_samples.cc topic-admin-client-set-endpoint
+@snippetgoogle/cloud/pubsub/admin/samples/topic_admin_client_samples.cc topic-admin-client-set-endpoint
 
 */
 
 /*! @page SubscriptionAdminClient-endpoint-snippet Override @c SubscriptionAdminClient Default Endpoint
 
-@snippet google/cloud/pubsub/samples/client_samples.cc subscription-admin-client-set-endpoint
+@snippet google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc subscription-admin-client-set-endpoint
```
Comment (dbolduc) [inline google/cloud/pubsub/doc/pubsub-main.dox:129]: this should point to the file for `subscription_admin_client_samples.cc`, not topic admin.

File: google/cloud/pubsub/samples/client_samples.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/pubsub/samples/client_samples.cc:163]: No, I would delete the Topic/Subscription admin client samples in this PR.

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: google/cloud/pubsub/samples/client_samples.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/client_samples.cc]: Done

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/doc/pubsub-main.dox]: I see now. Done.

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 3 above)
Comment (alevenberg) [reply google/cloud/pubsub/doc/pubsub-main.dox]: whoops done

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 4]
```diff
@@ -146,7 +146,7 @@ Follow these links to find examples for other `*Client` classes:
 
 /*! @page Subscriber-service-account-snippet Override @c Subscriber Default Authentication
 
-@snippet google/cloud/pubsub/samples/client_samples.cc subscriber-service-account
+@snippet google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc subscriber-service-account
```
[the remark this answers is not in the document]
Comment (alevenberg) [reply google/cloud/pubsub/doc/pubsub-main.dox]: Done

**Review by dbolduc — APPROVED** [review]
yay, less code.

Hold off on merging yadda yadda...

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 5]
```diff
@@ -120,13 +120,13 @@ Follow these links to find examples for other `*Client` classes:
 
 /*! @page TopicAdminClient-endpoint-snippet Override @c TopicAdminClient Default Endpoint
 
-@snippet google/cloud/pubsub/samples/client_samples.cc topic-admin-client-set-endpoint
+@snippet google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc topic-admin-client-set-endpoint
```
Comment (dbolduc) [inline google/cloud/pubsub/doc/pubsub-main.dox:123]: o, these things are just called `set-client-endpoint`, not `foo-client-set-endpoint`

https://github.com/googleapis/google-cloud-cpp/blob/3f8954c9c3f1394e969b2412be0a4ad776805314/google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc#L40

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 6]
```diff
@@ -158,13 +158,13 @@ Follow these links to find examples for other `*Client` classes:
...
 
-@snippet google/cloud/pubsub/samples/client_samples.cc topic-admin-client-service-account
+@snippet google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc topic-admin-client-service-account
 
 */
 
 /*! @page SubscriptionAdminClient-service-account-snippet Override @c SubscriptionAdminClient Default Authentication
 
-@snippet google/cloud/pubsub/samples/client_samples.cc subscription-admin-client-service-account
+@snippet google/cloud/pubsub/admin/samples/subscription_admin_client_samples.cc subscription-admin-client-service-account
```
Comment (dbolduc) [inline google/cloud/pubsub/doc/pubsub-main.dox:167]: https://github.com/googleapis/google-cloud-cpp/blob/3f8954c9c3f1394e969b2412be0a4ad776805314/google/cloud/pubsub/admin/samples/topic_admin_client_samples.cc#L112

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 5 above)
Comment (alevenberg) [reply google/cloud/pubsub/doc/pubsub-main.dox]: ack, done

**Pull request merged**